src/shell/xdg/frame/mod.rs

```diff
@@ -0,0 +1,116 @@
+//! The frame to use with XDG shell window.
+
+use crate::reexports::client::protocol::wl_surface::WlSurface;
+use crate::reexports::protocols::xdg::shell::client::xdg_toplevel::{
+    ResizeEdge, State as WindowState,
+};
+
+pub mod fallback_frame;
+
+/// The interface for the client side decorations.
+pub trait DecorationsFrame: Sized {
+    /// Configurations for this frame.
+    type Config;
+
+    /// Emulate click on the decorations.
+    ///
+    /// The `click` is a variant of click to use, see [`FrameClick`] for more info.
+    /// The return value is a [`FrameAction`] you should apply, this action could be
+    /// ignored, in case you don't want to apply it.
+    ///
+    /// The location of the click is the one passed to [`Self::click_point_moved`].
+    fn on_click(&mut self, click: FrameClick, pressed: bool) -> Option<FrameAction>;
+
+    /// Emulate pointer moved event on the decorations frame.
+    ///
+    /// The `x` and `y` are location in surface local coordinates relative to `surface`.
+    /// This function returns the new cursor icon you should apply to provide better visual
+    /// effect for the end user.
+    fn click_point_moved(&mut self, surface: &WlSurface, x: f64, y: f64) -> Option<&str>;
+
+    /// All clicks left the decorations.
```

Comment:
Clarify what was meant here?

Reply:
it's simply to move pointer away from the decoration frame.

Reply:
Isn't moving the pointer off the frame possible even without clicking? I do understand that clicking and the client not doing an interactive resize or move is possible.

I could see more advanced decorations possibly highlighting parts of the frame if the pointer went over some point if not clicking.

Reply:
>Isn't moving the pointer off the frame possible even without clicking?

Ah, guess comment is what actually confuses you? because it should probably be `all click points` left the decorations.

Be aware that I use `click` terminology to show that we're not talking about `wl_pointer` events in the first place, you could use all other sorts of inputs with it, like touch.

The use case for such API is to simply `leave` so `Hovered` state will go away, for example when clicking on minimize button and then restoring back.